server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3083,6 +3086,22 @@ private void cacheRepositoryData(BytesReference updated, long generation) {
         }
     }
 
+    protected long determineCacheSizeLimit() {
+        final String cacheSizeProperty = System.getProperty("opensearch.snapshot.cache.size");
+        long defaultCacheSizeLimit = ByteSizeUnit.KB.toBytes(500);
+
+        if (Strings.hasLength(cacheSizeProperty)) {
+            long maxHeapSize = Runtime.getRuntime().maxMemory();
+            double cacheSizeRatio = Double.parseDouble(cacheSizeProperty) / 100.0;
```

Comment:
1. Does this need exception handling?  The catch block in lines 3075-3078 won't catch this or give an appropriate error message.
2. This likely needs protection against values outside the range 0 - some maximum well below 100%
3. This seems to assume we're always giving a percentage. Is this consistent with other percentage-based property parsing?  IIRC we need the percentage character (like [this](https://github.com/opensearch-project/OpenSearch/blob/6f1b59e54bec41d40772f8571c7b65d4b523f8b1/server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdSettings.java#L65)) to assume this division by 100, and I'm assuming those likely have their own parsing method somewhere.

Reply:
Thank you for your helpful feedback. Based on your insights, I am currently revising the code.
Once [the discussion on the max limit](https://github.com/opensearch-project/OpenSearch/issues/16298#issuecomment-2444951524) is completed, I will incorporate that as well and push the updates again. Thank you for your patience!

As you suggested, I have also raised an issue on the documentation project. I appreciate all the guidance! 😄😆😁